lib/broadway_rabbitmq/producer.ex

```diff
@@ -271,6 +271,66 @@ defmodule BroadwayRabbitMQ.Producer do
   One way to handle this is by using [Dead Letter Exchanges](https://www.rabbitmq.com/dlx.html)
   and [TTL and Expiration](https://www.rabbitmq.com/ttl.html).
 
+  ### Dead letter exchange
+
+  #### Example
+
+      use Broadway
+
+      @exchange "x.my_exchange"
+      @exchange_dlx "x.my_exchange.dlx
+      @queue "x.my_queue"
+      @queue_dlx "x.my_queue.dlx"
+
+      def start_link(_opts) do
+        initialize_queues()
+
+        Broadway.start_link(__MODULE__,
+          name: MyDeadLetterBroadway,
+          producer: [
+            module: {
+              BroadwayRabbitMQ.Producer,
+              on_failure: :reject_and_requeue_once,
+              queue: @queue,
+              bindings: [{@exchange, []}],
+              declare: [
+                durable: true,
+                arguments: [
+                  {"x-dead-letter-exchange", :longstr, @exchange_dlx},
+                  {"x-dead-letter-routing-key", :longstr, @queue_dlx}
+                ]
+              ],
+            },
+            concurrency: 4
+          ],
+          processors: [
+            default: [
+              concurrency: 4,
+            ]
+          ]
+        )
+      end
+
+      defp initialize_queues do
```

Comment:
Is this initialization queues going to be fine if the queues already exist? Would it be like a no-op?

Reply:
Yes Rabbit declarations are idempotent so if these already exist then it's a no-op

Reply:
Yes, unless the exchanges/queues were previously declared with different configuration, i.e. `durable: false`.